modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -290,9 +294,104 @@ class ResizeSubgraph2 : public ExtractScalesSubgraph
+        int gather1 = addNodeToMatch("Gather", shape1, indexH);
+
+        int shape2 = addNodeToMatch("Shape", input2);
+        int indexW = addNodeToMatch("Constant");
+        int gather2 = addNodeToMatch("Gather", shape2, indexW);
+
+        int unsqueeze1 = addNodeToMatch("Unsqueeze", gather1);
+        int unsqueeze2 = addNodeToMatch("Unsqueeze", gather2);
+        concatId = addNodeToMatch("Concat", unsqueeze1, unsqueeze2);
+    }
+
+protected:
+    int input1, input2, shape, concatId;
+};
+
+class UpsampleTwoInputSubgraph : public UpsampleTwoInput
+{
+public:
+    UpsampleTwoInputSubgraph()
+    {
+        int constant = addNodeToMatch("Constant");
+        int cast1 = addNodeToMatch("Cast", concatId);
+
+        int slice = addNodeToMatch("Slice", shape);
+        int cast2 = addNodeToMatch("Cast", slice);
+
+        int div = addNodeToMatch("Div", cast1, cast2);
+        int concat = addNodeToMatch("Concat", constant, div);
+
+        addNodeToMatch("Upsample", input1, concat);
+        setFusedNode("Upsample", input1, input2);
+    }
+};
+
+class ResizeTwoInputSubgraph : public UpsampleTwoInput
+{
+public:
+    ResizeTwoInputSubgraph()
+    {
+        int slice = addNodeToMatch("Slice", shape, addNodeToMatch("Constant"), addNodeToMatch("Constant"), addNodeToMatch("Constant"));
+
+        int castConcat = addNodeToMatch("Cast", concatId);
+        int concat = addNodeToMatch("Concat", slice, castConcat);
+        int constant = addNodeToMatch("Constant");
+
+        addNodeToMatch("Resize", input1, constant, constant, concat);
+        setFusedNode("Upsample", input1, input2);
+    }
+};
+
+class BatchNormalizationSubgraph : public Subgraph
+{
+public:
+    BatchNormalizationSubgraph()
+    {
+        int input = addNodeToMatch("");
+        int data1 = addNodeToMatch("Initializer");
+        int data2 = addNodeToMatch("Initializer");
+        int data3 = addNodeToMatch("Initializer");
+        int data4 = addNodeToMatch("Initializer");
+        int shape1 = addNodeToMatch("Constant");
+        int reshape1 = addNodeToMatch("Reshape", data1, shape1);
+        int shape2 = addNodeToMatch("Constant");
+        int reshape2 = addNodeToMatch("Reshape", data2, shape2);
+        int shape3 = addNodeToMatch("Constant");
+        int reshape3 = addNodeToMatch("Reshape", data3, shape3);
+        int shape4 = addNodeToMatch("Constant");
+        int reshape4 = addNodeToMatch("Reshape", data4, shape4);
+        int sqrt = addNodeToMatch("Sqrt", reshape3);
+        int A = addNodeToMatch("Constant");
+        int div = addNodeToMatch("Div", A, sqrt);
+        int mul1 = addNodeToMatch("Mul", reshape1, div);
+        int mul2 = addNodeToMatch("Mul", reshape4, mul1);
+        int sub = addNodeToMatch("Sub", reshape2, mul2);
+        int mul3 = addNodeToMatch("Mul", input, mul1);
+        addNodeToMatch("Add", mul3, sub);
+        setFusedNode("BatchNormalization", input, data1, data2, data4 ,data3);
+    }
+};
```

Comment:
I'd like to recommend alternative solution which will work for other cases with `"Initializer"` nodes (i.e. as this resize https://github.com/daquexian/onnx-simplifier/issues/34). Before the fusion, create copies of the Initializer contents as regular Constant nodes. (i.e. as the first graph nodes).

Reply:
Yeah, that can also be done. I try.

Reply:
@dkurt I just want to make sure that you mean before simplify subgraph,
for each node: 
replace initializer input by constant node (created by copying initializer tensor data) input so that number of inputs for each node will be same as before.

Reply:
That's right! Otherwise we will have to duplicate existing subgraphs fusion to manage both Constant and Initializer while then look similar as for me.

Reply:
@dkurt I have made the suggested changes. Just added constant nodes with the same output id as initializers so that Graph matching can be done without matching initializers.

Reply:
Thanks!